WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/payment/BlazeCampaignPaymentMethodsListViewModel.kt

```diff
@@ -45,8 +51,32 @@ class BlazeCampaignPaymentMethodsListViewModel @Inject constructor(
     )
 
     private fun addPaymentMethodWebView(): ViewState = ViewState.AddPaymentMethodWebView(
-        urls = navArgs.paymentMethodsData.addPaymentMethodUrls,
-        onUrlLoaded = { /* TODO */ },
+        formUrl = navArgs.paymentMethodsData.addPaymentMethodUrls.formUrl,
+        userAgent = userAgent,
+        wpComWebViewAuthenticator = wpComWebViewAuthenticator,
+        onUrlLoaded = { url ->
+            if (url.startsWith(navArgs.paymentMethodsData.addPaymentMethodUrls.successUrl)) {
+                runCatching {
+                    URL(url).query?.split("&")
+                        ?.firstOrNull { it.startsWith("payment_method_id=") }
+                        ?.substringAfter("=")
+                        .let { requireNotNull(it) }
+                }.fold(
+                    onSuccess = { paymentMethodId ->
+                        triggerEvent(
+                            MultiLiveEvent.Event.ShowSnackbar(
+                                R.string.blaze_campaign_payment_added_successfully
+                            )
+                        )
+                        MultiLiveEvent.Event.ExitWithResult(paymentMethodId)
```

Comment:
This is probably an accidental change. Missing the triggering of this event for the webview to be dismissed:
```suggestion
                        triggerEvent(MultiLiveEvent.Event.ExitWithResult(paymentMethodId))
```

Reply:
🤦, thanks @JorgeMucientes for catching this.